### ISSUE TITLE
Constructor type being linked to property

### ISSUE THREAD

**Issue by Levi-Lesches** [dart-lang/dartdoc#2007]
Consider the following:
```dart
class Temp {
  final int temp;
  
  Temp (this.temp);
}
```

In the docs for this class, clicking on the `Temp` of the constructor links to `temp` the property. Is there any way to get around this?

**Comment by Levi-Lesches** [comment]
Also, searching it gives "Temp (constructor)" but clicking on the search result gives `temp` the property.

**Comment by keertip** [comment]
Which OS are you running on? This is likely due to the fact that the file system is not case sensitive and so the documentation for `Temp` constructor is overwritten by doc for `temp` property, the file names being the same expect for `T` vs 't'. Try renaming `temp` and see if this is still an issue.

**Comment by Levi-Lesches** [comment]
Ah that makes sense. And yeah, in my many other classes this issue is not present (except for #2008, but I doubt this is related). But even if my host OS supported a case-sensitive file system, wouldn't anyone who downloaded it have problems viewing it? How can this be avoided in general? Can dartdoc change the filenames like `Temp-constructor.html` and `temp-property.html`?

**Comment by keertip** [comment]
There was a similar bug where we changed the file names [#585](https://github.com/dart-lang/dartdoc/issues/585).

**Comment by Levi-Lesches** [comment]
And if #1983 is being worked on, maybe this whole thing can be avoided? The general goal of it seems to be to no longer need a separate file for every member of a class, so that should help.

**Comment by keertip** [comment]
Yes,  #1983 would address this issue, but there is no eta on when it will be worked on.

**Comment by Levi-Lesches** [comment]
So what should be done in the meantime? Should I just rename properties going forward to avoid this?

**Comment by jcollins-g** [comment]
This is a direct duplicate of #1196 though as @keertip mentions #1983 would likely fix this as well.

Renaming properties or using a case-sensitive filesystem are the only workarounds we have at the moment.
